content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+
+Pod Security Admission also adheres to Kubernetes best practices of declarative object management by denying resources that violate the policy. This requires resources to be updated in source repositories, and tooling to be updated prior to being deployed to Kubernetes.
+
+## How Does Pod Security Admission Work?
+    
+Pod Security Admission is a built-in [admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/). Admission controllers function by intercepting requests in the Kubernetes API server prior to persistence to storage. They can either `admit` or `deny` a request. In the case of Pod Security Admission, pod specifications will be evaluated against a configured policy in the form of a Pod Security Standard. This means that security sensitive fields in a pod specification will only be allowed to have [specific](https://kubernetes.io/docs/concepts/security/pod-security-standards/#profile-details) values. 
+
+## Configuring Pod Security Admission
+    
+### Pod Security Standards
+    
+In order to use Pod Security Admission we first need to understand [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/). These standards define three different policy levels that range from permissive to restrictive. These levels are as follows:
+   * Privileged — open and unrestricted
+   * Baseline — Covers most common known privilege escalations and provides an easier onboarding
+   * Restricted — Highly restricted, covering best practices. May cause compatibility issues
+
+Each of these policy levels define which fields are restricted within a pod specification and the allowed values. Some of the fields restricted by these policies include:
+   * `spec.containers[*].ports`
+   * `spec.volumes[*].hostPath`
+   * `spec.securityContext`
+   * `spec.containers[*].securityContext`
+    
+These policy levels can be applied in two different ways. Either via namespace labels or an AdmissionConfiguration resource. Using namespace labels allows for granular per namespace policy selection whereas using AdmissionConfiguration allows cluster-level defaulting along with exemptions.
+
+### Policy modes
+
+Policies are applied using modes. Here is a list of modes:
+   * `enforce` — Any Pods that violate the policy will be rejected
+   * `audit` — Pods with violations will be allowed and an audit annotation will be added
+   * `warn` — Pods that violate the policy will be allowed and a warning message will be sent back to the user.
+
+In addition to modes you can also pin the policy to a specific version for example v1.22. Pinning to a specific version allows the behavior to remain consistent as policy changes happen over Kubernetes releases.
+
+## Hands on demo
```

Comment:
Can instead
- link to https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-admission-controller/ instead?
- copy the steps from there?

_or_ merge these steps into that task page?

I don't want us to have two pages that document different ways to try the same thing, on release day. That'd feel counter-productive as readers might wonder which they should follow.

Reply:
You could also add this content within https://docs.k8s.io/tutorials/ - I think that'd be a really good choice.

Reply:
@sftim do you think the additional content should be included with the blog PR or a follow-up? Happy to open a tracking issue. Also, are you saying to link the entire hands-on demo portion to the official docs and move the demo there?

I like both ideas, if I understand them correctly, but it might be worth tracking vs. implementing.